src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/BackupsService.java

```diff
@@ -0,0 +1,190 @@
+    /**
+     * Return a list of available backup files
+     * @return
+     */
+    public List<Backup> getBackupFiles() {
+        List<File> backupFiles = backupsRepository.getBackupFiles();
+        List<Backup> backups = backupsConverter.convert(backupFiles);
+        setCreatedAt(backups);
+        return backups;
+    }
+
+    /**
+     * Delete a backup file or throws an exception if the file does not exist
+     * @param filename
+     * @throws BackupFileNotFoundException
+     */
+    public void deleteBackup(String filename) throws BackupFileNotFoundException {
+        if (!backupExists(filename).isPresent()) {
+            throw new BackupFileNotFoundException(getFileNotFoundExceptionMessage(filename));
+        }
+        backupsRepository.deleteBackupFile(filename);
+    }
+
+    /**
+     * Read backup file's content
+     * @param filename
+     * @return
+     * @throws BackupFileNotFoundException
+     */
+    public byte[] readBackupFile(String filename) throws BackupFileNotFoundException {
+        if (!backupExists(filename).isPresent()) {
+            throw new BackupFileNotFoundException(getFileNotFoundExceptionMessage(filename));
+        }
+        return backupsRepository.readBackupFile(filename);
+    }
+
+    /**
+     * Generate a new backup file
+     * @throws ProcessFailedException
+     * @throws ProcessNotExecutableException
+     * @throws InterruptedException
+     * @return
+     */
+    public Backup generateBackup() throws InterruptedException, BackupFileNotFoundException {
+        SecurityServerId securityServerId = serverConfService.getSecurityServerId();
+        String filename = generateBackupFileName();
+        String fullPath =  backupsRepository.getConfigurationBackupPath() + filename;
+        String[] args = new String[] {"-s", securityServerId.toShortString(), "-f", fullPath};
+
+        try {
+            log.info("Run configuration backup with command '"
+                    + generateBackupScriptPath + " " + Arrays.toString(args) + "'");
+
+            List<String> output = externalProcessRunner.execute(generateBackupScriptPath, args);
+
+            log.info(" --- Backup script console output - START --- ");
+            log.info(String.join("\n", output));
+            log.info(" --- Backup script console output - END --- ");
+        } catch (ProcessNotExecutableException | ProcessFailedException e) {
+            log.error("Failed to generate backup", e);
+            throw new DeviationAwareRuntimeException(e, new ErrorDeviation(BACKUP_GENERATION_FAILED));
+        }
+
+        Optional<Backup> backup = backupExists(filename);
+        if (!backup.isPresent()) {
+            throw new BackupFileNotFoundException(getFileNotFoundExceptionMessage(filename));
+        }
+        return backup.get();
+    }
+
+    /**
+     * Check if a backup file with the given filename exists
+     * @param filename
+     * @return
+     */
+    private Optional<Backup> backupExists(String filename) {
```

Comment:
maybe more idiomatic naming is that a `backupExists()` method returns boolean, and method returning Optional is named e.g. `getBackup()`

Reply:
Method name updated.